core/store/orm/orm.go

```diff
@@ -613,6 +622,22 @@ func (orm *ORM) UpdateTx(
 	return orm.DB.Save(tx).Error
 }
 
+// FindLaterConfirmedTx finds the earliest confirmed Ethereum transaction
+// that has the same sender and a higher nonce.
+func (orm *ORM) FindLaterConfirmedTx(tx *models.Tx) (*models.Tx, error) {
+	later := models.Tx{}
+	rval := orm.DB.Order("nonce asc").
+		Where("\"confirmed\" = true AND \"from\" = ? AND \"nonce\" > ?", tx.From, tx.Nonce).
+		First(&later)
```

Comment:
Why not just do `.Error` and check `if errors.Cause(err)`?

Reply:
Because we want to catch and ignore `RecordNotFound` errors, but rval can have multiple errors so we have a helper to do that.

Reply:
`First` can only have one error, and is checked with `First(...).Error`.